calculator/src/Calculator.jsx

```diff
@@ -0,0 +1,153 @@
+import { useEffect, useState } from "react";
+import "./css/index.css";
+import {
+  FONT_SIZE_STANDARD,
+  INFINITY_MESSAGE,
+  MAX_CURRENT_LENGTH,
+  digits,
+  operators,
+} from "./constants";
+
+const Calculator = () => {
+  const [total, setTotal] = useState(0);
+  const [current, setCurrent] = useState(0);
+  const [operator, setOperator] = useState("");
+  const [isLastClickOperator, setIsLastClickOperator] = useState(false);
+
+  useEffect(() => {
+    const calcInfo = localStorage.getItem("calcInfo");
+    if (calcInfo) {
+      const { total, current, operator, isLastClickOperator } =
+        JSON.parse(calcInfo);
+      setTotal(total);
+      setCurrent(current);
+      setOperator(operator);
+      setIsLastClickOperator(isLastClickOperator);
+    }
+  }, []);
+  useEffect(() => {
+    return () => {
+      localStorage.setItem(
+        "calcInfo",
+        JSON.stringify({ total, current, operator, isLastClickOperator })
+      );
+    };
+  }, [total, current, operator, isLastClickOperator]);
+
+  const handleDigitClick = (digitValue) => {
+    if (current.toString().length > MAX_CURRENT_LENGTH) {
+      setCurrent(Infinity);
+      return;
+    }
+
+    if (isLastClickOperator) {
+      enterNewOperand(digitValue);
+      return;
+    }
+
+    concatDigitOperand(digitValue);
+  };
+
+  const enterNewOperand = (digitValue) => {
+    setCurrent(digitValue);
+    setIsLastClickOperator(false);
+  };
+
+  const concatDigitOperand = (digitValue) => {
+    setCurrent((prevState) => prevState * 10 + digitValue);
+  };
+
+  const handleOperatorClick = (operatorValue) => {
+    updateOperator(operatorValue);
+
+    if (isLastClickOperator) return;
+
+    if (!operator) {
+      calculate(current);
+      return;
+    }
```

Comment:
아래와 같이 수정하면 된다면 updateOperator 함수에서 setOperator를 할 필요가 없어질것 같은데 어떠세요?
```suggestion
    if (operator !== "=") {
      calculate(current);
      return;
    }
```

Reply:
추가로 setOperator("")으로 해주는 부분도 수정이 필요합니다!